src/iot/commands/deleteThing.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as localizedText from '../../shared/localizedText'
+import { getLogger } from '../../shared/logger'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { Commands } from '../../shared/vscode/commands'
+import { Window } from '../../shared/vscode/window'
+import { IotThingNode } from '../explorer/iotThingNode'
+import { showViewLogsMessage, showConfirmationMessage } from '../../shared/utilities/messages'
+import { IotThingFolderNode } from '../explorer/iotThingFolderNode'
+
+/**
+ * Deletes the thing represented by the given node.
+ *
+ * Prompts the user for confirmation.
+ * Deletes the thing.
+ * Refreshes the parent node.
+ */
+export async function deleteThingCommand(
+    node: IotThingNode,
+    window = Window.vscode(),
+    commands = Commands.vscode()
+): Promise<void> {
+    getLogger().debug('DeleteThing called for %O', node)
+
+    const thingName = node.thing.name
+
+    const isConfirmed = await showConfirmationMessage(
+        {
+            prompt: localize('AWS.iot.deleteThing.prompt', 'Are you sure you want to delete Thing {0}?', thingName),
+            confirm: localizedText.localizedDelete,
+            cancel: localizedText.cancel,
+        },
+        window
+    )
+    if (!isConfirmed) {
+        getLogger().info('DeleteThing canceled')
+        return
+    }
+
+    getLogger().info(`Deleting thing ${thingName}`)
+    try {
+        const principalList = (await node.iot.listThingPrincipals({ thingName: thingName })).principals
+        if (principalList?.length ?? 0 > 0) {
+            getLogger().error(`Thing ${thingName} has attached principals: %O`, principalList)
```

Comment:
Do you mean this information should be in the `showViewLogsMessage` instead of just the `getLogger().error()`, or that `showViewLogsMessage` shouldn't be used at all, and something like `window.showErrorMessage` should be used instead?

Reply:
Probably something like `showErrorMessage` or `showWarningMessage`. Essentially, this is not an arbitrary error and is probably one that's best immediately surfaced in front of a user so they know to action it. Logs should be used for diagnostic information (you can add the message to the logs too) but there's no point in having them round trip to the logs in order to figure this one out. I think there's a couple other locations where you have similar messages being written to logs.

Reply:
I replaced all of the predictable `showViewLogsMessage`s (the ones not in a `catch (e)` block) with `showErrorMessage`